shared/util/saga.js

```diff
@@ -0,0 +1,11 @@
+// @flow
+
+import {put, cps} from 'redux-saga/effects'
+
+export function * cpsWithWaiting (waitingFn: (isWaiting: boolean) => any, cpsFn: any, ...args: any): any {
+  yield put(waitingFn(true))
```

Comment:
is it better to assume waitingFn returns an action vs just having it be a generic function we call. we don't necessarily care what it does? i imagine more like

```
yield call(waitingFn, true)
```

and then

```
function waiting (waiting: boolean): void {
   yield put({type: Constants.waiting, payload: {waiting}})
 }
```

??